src/security_utilities_rust/src/microsoft_security_utilities_core/identifiable_secrets.rs

```diff
@@ -811,3 +814,114 @@ fn index_of(input_string: String, search_char: char) -> i32
     }
 }
 
+#[derive(Clone)]
+struct Detection {
+    start: u64,
+    end: u64,
+    length: usize,
+    redaction_token: String
+}
+
+pub struct SecretMasker {
+    pub scan: identifiable_scans::Scan
+}
+
+impl SecretMasker {
+    pub fn mask_secrets(&mut self, input: &str, default_redaction_token: Option<&str>, validate_checksum: bool) -> String {
+        if input.is_empty() {
+            return String::new();
+        }
+
+        self.scan.reset();
+
+        let input_as_bytes = input.as_bytes();
+        self.scan.parse_bytes(input_as_bytes);
+
+        let detections = self.scan.possible_matches();
+        let mut detections = detections.clone();
+
+        // Short-circuit if nothing to replace.
+        if detections.is_empty() {
+            return input.to_string();
```

Comment:
This causes an allocation and string copy. When there are no secrets, we want that to be fast, right?

Reply:
I would suggest the function returning an Option<String> if you want to not have in-place masking.

Reply:
Valid suggestion. Fixed.